functions/assert_metadata.pp

```diff
@@ -25,8 +27,9 @@
 function simplib::assert_metadata (
   String[1] $module_name,
   Optional[Struct[{
-    enable => Optional[Boolean],
-    os     => Optional[Struct[{
+    enable    => Optional[Boolean],
+    blacklist => Optional[Array[String]],
+    os        => Optional[Struct[{
```

Comment:
Should `blacklist` be part of the `os` structure instead of a separate top-level array?  That way it could take advantage of the existing logic, and fail on `none`, `full`, `major`.

Reply:
I was thinking about this but it would basically have to be another identical hash.